students/johnpharmd/lesson05/mailroom3.py

```diff
@@ -183,28 +197,28 @@ def get_report():
     for i in range(55):
         print('-', end='')
     print()
-    # add comprehension here
+    # add comprehension here?
     for donor in donors_amts:
-        d1 = donors_amts[donor][1]
-        d2 = donors_amts[donor][2]
+        d1 = donors_amts[donor]['donations']
+        d2 = donors_amts[donor]['num_of_donations']
         print('{:<15}{}{:>10}{:>12}{}{:>11}'.format(donor, '  $',
               d1, d2, '  $', d1 // d2))
     print()
     program_run()
```

Comment:
your `program_run` needs to have a while True loop, then you wouldn't need to call this from every function.